frontend/components/TeamsDropdown/TeamsDropdown.tests.tsx

```diff
@@ -0,0 +1,82 @@
+import React from "react";
```

Comment:
This looks great! I can't remember where the source of truth to all this logical is so I'm not quite sure if we're missing any edge cases.

Might make a good integration test to do this for all the different RBAC and pages.

Reply:
i based these tests on how code for the component works so they should cover most of the common use cases, but always good to add more edge cases. Lets do that in a future tech debt PR.

Plus I wanna think about a good plan for how we should approach RBAC tests as we're likely missing many throughout the UI